torch/csrc/cuda/Module.cpp

```diff
@@ -150,9 +150,8 @@ PyObject * THCPModule_setStream_wrap(PyObject *self, PyObject *obj)
 
 PyObject * THCPModule_isDriverSufficient(PyObject *self, PyObject *noargs)
 {
-  int count;
-  cudaError_t err = cudaGetDeviceCount(&count);
-  if (err == cudaErrorInsufficientDriver) {
+  const auto count = c10::cuda::device_count();
+  if (count == 0) {
```

Comment:
that's actually wrong and will produce the misleading error message if driver is available but there are no GPUs on the system (see https://www.internalfb.com/intern/diffusion/FBS/browsefile/master/xplat/caffe2/torch/cuda/__init__.py?lines=62)

I can probably move more of this logic to C++ as I started moving some in https://github.com/pytorch/pytorch/pull/42249/files